filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/batch/GcsBatchIoCommand.scala

```diff
@@ -174,23 +176,39 @@ object GcsBatchSizeCommand {
     file.objectBlobId.map(GcsBatchSizeCommand(file, _))
 }
 
-case class GcsBatchCrc32Command(override val file: GcsPath, override val blob: BlobId, setUserProject: Boolean = false)
-    extends IoHashCommand(file)
+case class GcsBatchHashCommand(override val file: GcsPath,
+                               override val hashStrategy: FileHashStrategy,
+                               val blob: BlobId,
+                               setUserProject: Boolean = false
+) extends IoHashCommand(file, hashStrategy)
     with GcsBatchGetCommand[String] {
-  override def mapGoogleResponse(response: StorageObject): ErrorOr[String] =
-    Option(response.getCrc32c) match {
-      case None => s"'${file.pathAsString}' in project '${file.projectId}' returned null CRC32C checksum".invalidNel
-      case Some(crc32c) => crc32c.validNel
+  override def mapGoogleResponse(response: StorageObject): ErrorOr[String] = {
+    val fileHash = hashStrategy
+      .getFileHash(
+        response,
+        (resp: StorageObject, hashType: HashType) =>
+          hashType match {
+            case HashType.Crc32c => Option(response.getCrc32c)
+            case HashType.Md5 => Option(response.getMd5Hash)
+            case HashType.Identity => Option(response.getId)
```

Comment:
All my waffling about whether to make Identity a separate strategy resulted in my losing track of the need to md5sum the blob id, change incoming for this location and the similar one in NioHashing.